File: english-hub-backend/src/main/java/com/example/englishhubbackend/service/UserService.java
package com.example.englishhubbackend.service;

import com.example.englishhubbackend.dto.request.UserCreateRequest;
import com.example.englishhubbackend.dto.request.UserUpdateRequest;
import com.example.englishhubbackend.dto.response.UserResponse;

import java.util.List;
import java.util.UUID;

public interface UserService {
    boolean existsByEmail(String email);
    UserResponse deactivateUser(UUID userId);
    UserResponse activateUser(UUID userId);
    List<UserResponse> getAllUsers();
    UserResponse createUser(UserCreateRequest userCreateRequest);
    UserResponse updateUser(UUID userId, UserUpdateRequest userUpdateRequest);
}

File: english-hub-frontend/src/services/userService.ts
import axiosInstance from '@/services/axiosInstance';
import {
  UserCreateRequest,
  UserResponse,
  UserUpdateRequest,
} from '@/types/userType';

export async function getAllUsers(): Promise<UserResponse[]> {
  const response = await axiosInstance.get('/users');
  return response.data.result;
}

export async function createUser(
  user: UserCreateRequest
): Promise<UserResponse> {
  const response = await axiosInstance.post('/users', user);
  return response.data.result;
}

export async function updateUser(
  id: string,
  user: UserUpdateRequest
): Promise<UserResponse> {
  const response = await axiosInstance.put(`/users/${id}`, user);
  return response.data.result;
}

export async function deactivateUser(id: string): Promise<UserResponse> {
  const response = await axiosInstance.patch(`/users/${id}/deactivate`);
  return response.data.result;
}

export async function activateUser(id: string): Promise<UserResponse> {
  const response = await axiosInstance.patch(`/users/${id}/activate`);
  return response.data.result;
}

export async function getUserById(id: string): Promise<UserResponse> {
  const response = await axiosInstance.get(`/users/${id}`);
  return response.data.result;
}
